fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java

```diff
@@ -340,6 +341,16 @@ public Void visitAlterSystemStmt(AlterSystemStmt statement, ConnectContext sessi
 
         }
 
+        @Override
+        public Void visitAlterUserStatement(AlterUserStmt statement, ConnectContext context) {
+            // check if current user has GRANT priv on GLOBAL level.
+            if (!GlobalStateMgr.getCurrentState().getAuth().checkGlobalPriv(
+                    ConnectContext.get(), PrivPredicate.GRANT)) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "GRANT");
```

Comment:
if (!GlobalStateMgr.getCurrentState().getAuth()
                .checkHasPriv(ConnectContext.get(), PrivPredicate.GRANT, Auth.PrivLevel.GLOBAL,
                        Auth.PrivLevel.DATABASE)) {
            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "GRANT");
        }

miss database privilege?

Reply:
you 're right, added now